lib/chef/provider/package.rb

```diff
@@ -148,6 +151,7 @@ def upgrade_description
       private :upgrade_description
 
       action :remove do
+        validate_package(new_resource.package_name)
```

Comment:
not quite sure why we're still calling this directly in :remove/:purge/:reconfig when those actions all call DRR as well.

Reply:
Hi @lamont-granquist  I have done with above changes also DRR is called only for :install/upgrade for :remove/:purge/:reconfig we are not calling that methods. I am testing on the changes and will update for the same.